gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.groovy

```diff
@@ -32,18 +33,24 @@ class BaselineErrorProne extends AbstractBaselinePlugin {
             errorprone "com.palantir.baseline:baseline-error-prone:${extractVersionString()}"
         }
 
+        def errorProneConf = project.getConfigurations().getByName("errorprone")
         project.afterEvaluate { p ->
             p.tasks.withType(JavaCompile) {
                 options.compilerArgs += [
-                        "-Xbootclasspath/p:${project.getConfigurations().getByName("errorprone").getAsPath()}",
+                        "-Xbootclasspath/p:${errorProneConf.getAsPath()}",
                         "-XepDisableWarningsInGeneratedCode",
                         "-Xep:EqualsHashCode:ERROR",
                         "-Xep:EqualsIncompatibleType:ERROR",
                 ]
             }
 
             p.tasks.withType(Test) {
-                jvmArgs "-Xbootclasspath/p:${project.getConfigurations().getByName("errorprone").getAsPath()}"
+                jvmArgs "-Xbootclasspath/p:${errorProneConf.getAsPath()}"
+            }
+
+            p.tasks.withType(Javadoc) {
+                options.bootClasspath.addAll(errorProneConf.resolve())
+                options.bootClasspath.addAll(System.properties["sun.boot.class.path"].split(File.pathSeparator).collect{new File(it)})
```

Comment:
can we put some explanatory comments around this stuff? Perhaps just a note to say why it's necessary here or where it came from? I wouldn't really feel confident to touch any of this stuff myself for fear of breaking people